### PR TITLE
Use XmlSerializer instead of insecure BinaryFormatter in TransferDataSource

### PR DESCRIPTION
**Pull request by sevoku**

### CODE REVIEW

File: Xwt/Xwt/TransferDataSource.cs [hunk 1]
```diff
@@ -138,32 +138,25 @@ public object GetValue (TransferDataType type)
...
-			}
+
+        /// <summary>
+        /// Serializes a value to a byte array using <see cref="System.Text.Json.JsonSerializer"/> .
+        /// </summary>
+        /// <returns>The serialized value.</returns>
+        /// <param name="val">The value to serialize.</param>
+        public static byte[] SerializeValue (object val, Type type)
+        {
+            return JsonSerializer.SerializeToUtf8Bytes (val, type);
```
Comment (mrward) [inline Xwt/Xwt/TransferDataSource.cs:149]: tabs vs spaces?

Comment (sevoku) [reply Xwt/Xwt/TransferDataSource.cs]: fixed

**Review by mrward — APPROVED** [review]

**Comment by Therzok** [comment]
Can we instead download gtk# from the github releases? https://github.com/mono/gtk-sharp/releases

**Comment by sevoku** [comment]
> Can we instead download gtk# from the github releases? https://github.com/mono/gtk-sharp/releases

Oh nice, I didn't know this existed! Will try in https://github.com/mono/xwt/pull/1115

**Comment by Therzok** [comment]
> Oh nice, I didn't know this existed! Will try in #1115

Nor did I. I think it's behind a few versions, but it's newer than what you're installing in appveyor.

**Review by BretJohnson — APPROVED** [review]

**Comment by BretJohnson** [comment]
I updated this to use XmlSerializer instead of JsonSerializer, as that works better for Android designer drag & drop / clipboard (the formatting is better and in general the XML serializer is more mature, with more options). I also switched to .NET 4.7.2, from 4.6.1, on Windows. With those changes, I tested and (with other required changes on the Android designer side) everything works OK. So this is good to merge from my perspective.

File: Xwt/Xwt.csproj [hunk 2]
```diff
@@ -1,5 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (mrward) [inline Xwt/Xwt.csproj:1]: Minor: SDK style projects do not usually have this xml header.

**Review by mrward — APPROVED** [review]

**Review by Therzok — APPROVED** [review]

**Pull request merged**